k8s/core/configmap/configmap_lock_v2_test.go

```diff
@@ -402,3 +402,72 @@ func TestCMLockTTL(t *testing.T) {
 	require.Equal(t, owner, lockOwners[key])
 	require.Equal(t, currTime.Add(testLockTTL), lockExpirations[key])
 }
+
+func TestIsKeyLocked(t *testing.T) {
+	setUpConfigMapTestCluster(t)
+	cmIntf, err := New("px-configmaps-lock-is-key-locked-test", nil, testLockTimeout, testLockAttempts, testLockRefreshDuration, testLockTTL)
+	require.NoError(t, err, "Unexpected error on New")
+	cm := cmIntf.(*configMap)
+
+	id1 := "is-key-locked-id1"
+	id2 := "is-key-locked-id2"
+	key := "is-key-locked-key"
+
+	locked, _, err := cm.IsKeyLocked(key, id1)
+	require.NoError(t, err)
+	require.False(t, locked)
+
+	// lock the key
+	err = cm.LockWithKey(id1, key)
+	require.NoError(t, err)
+	require.True(t, cm.kLocksV2[key].refreshing)
```

Comment:
Map access outside of lock might throw concurrent map access error. Which could make this test flaky.